pages/blog/miniseries-part_1.md

```diff
@@ -0,0 +1,227 @@
+
+First we have the definition of **parameters** used in the channel. **serverId** tells us where the action originates from, the **playerId** tells us who performed the action, and the **itemId** tells us which item was picked up and should all validate against a value with type **string**.
+
+![Game server setup](/img/posts/jonaslagoni-miniseries-part1/blog-miniseries-gameserver-api.webp)
+
+Next we have the **subscribe** operation, which might not make much sense at first glance. We do want the **game server** to publish this event right? 
+
+And you would be correct, but this is how you currently define operations in AsyncAPI. You define the operation others may interact. This means that the **game server** publishes on this channel and others may **subscribe** to it \[[1](#view-property)\]\[[3](#clarify-view)\]. 
+
+The **payload** of the channel (is described using a super-set of JSON Schema draft 7) should validate against an **object** which contains the property **pickupTimestamp**, which should validate against a **string**. When **additionalProperties** is **false**, no extra properties may be added to the object (by default this is **true** in JSON Schema draft 7). The **$id** keyword are used as an identifier for that specific schema, in this case we name the object schema **PlayerItemPickupPayload**.
+
+## The backend processor
+Next we design the **processor** API which contains all the same channels as the **game server**, but with a different operation keyword. 
+
+![Processor setup](/img/posts/jonaslagoni-miniseries-part1/blog-miniseries-processor-api.webp)
+
+This is again because we want to define how others may interact with our **processor**. This means that instead of using the `subscribe` operation we use `publish` to tell others that they can publish to this channel since the backend **processor** are subscribing to it. The full AsyncAPI document for the **processor** can be found [here](https://github.com/jonaslagoni/asyncapi-miniseries/blob/master/AsyncAPI/Processor.yaml). 
+
+```yaml
+...
+  game/server/{serverId}/events/player/{playerId}/item/{itemId}/pickup:
+    ...
+    publish: 
+      ...
+...
+```
+
+## Introducing reusability
+At the moment each of the AsyncAPI documents contains their own definition of the channels. But what if we where to add a new validation rule such as a new property to the **playerItemPickupPayload** schema? In our case we would have to change this for both applications which is way too much work :smile:
+
+We can therefore introduce **$ref** to separate the parameters and messages into smaller sections for reusability. I will be placing all separate components into a **components** directory in the same directory the AsyncAPI documents reside.
+
+Just a quick note, at the moment it is not possible to reuse channels and operations directly, therefore we must do it to the parameters and message individually \[[2](#channel-reusability)\]. 
```

Comment:
1. I think the link is broken
2. Channel can be also reused with ref -> https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md#channel-item-object, which means technically it can be reused. I guess you mean that you cannot ref the operation, to have channel A with subscribe referencing operation that is also in channel B with publish, right? if yes, it must have better explanation or just remove this sentence 😄 to not put to much knowledge on the reader.

Reply:
Which link are you referring to? 

True, I will clarify it some more 👍

Reply:
forget about the link, I did not notice that you use markdown references. I don't really like them, prefer link in the text, but it is up to you

Reply:
Hmm, let me sleep on it 😄